pkg/install/generate.go

```diff
@@ -615,3 +648,64 @@ func completeOpenStackDeprovisionJob(req *hivev1.ClusterDeprovision, job *batchv
 	job.Spec.Template.Spec.Containers = containers
 	job.Spec.Template.Spec.Volumes = volumes
 }
+
+func completeVSphereDeprovisionJob(req *hivev1.ClusterDeprovision, job *batchv1.Job) {
+	volumes := []corev1.Volume{}
+	volumeMounts := []corev1.VolumeMount{}
+	credentialsSecret := ""
+	if len(req.Spec.Platform.VSphere.CredentialsSecretRef.Name) > 0 {
+		credentialsSecret = req.Spec.Platform.VSphere.CredentialsSecretRef.Name
+	}
+	env := []corev1.EnvVar{}
+	if len(credentialsSecret) > 0 {
+		env = append(
```

Comment:
Extract the code for generating the environment variables from the creds secret into a method so that it can be shared by the install and uninstall paths.